hazelcast/src/main/java/com/hazelcast/mapreduce/aggregation/AbstractSupplyingMapper.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.mapreduce.aggregation;
+
+import com.hazelcast.mapreduce.Context;
+import com.hazelcast.mapreduce.Mapper;
+import com.hazelcast.mapreduce.aggregation.impl.SimpleEntry;
+
+/**
+ * Base class for mapper implementations used in aggregations.
```

Comment:
it would be nice to have a description why to use this rather than implementing the interface directly. 
What does it give me?